src/MonitorProfiler/MainProfiler/MainProfiler.cpp

```diff
@@ -19,6 +19,14 @@ using namespace std;
 
 #define IfFailLogRet(EXPR) IfFailLogRet_(m_pLogger, EXPR)
 
+#ifndef DLLEXPORT
+#define DLLEXPORT
+#endif
+
+typedef INT32 (STDMETHODCALLTYPE *ManagedMessageCallback)(INT16, const BYTE*, UINT64);
+std::mutex g_messageCallbackMutex;
```

Comment:
Let's make these static but class scoped at least? Random globals are difficult to work with.